File: python/15650.py
n, m = map(int, input().split())

arr = [0 for _ in range(8)]
visited = [False for _ in range(8)]


def foo(idx, cnt):
    if cnt == m:
        for i in range(m):
            print(arr[i], end=" ")
        print("")
        return

    for i in range(idx , n):
        if not visited[i]:
            visited[i] = True
            arr[cnt] = i + 1
            foo(i+1, cnt + 1)
            visited[i] = False


foo(0, 0)

File: javascript/15650.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.on('line', (line) => {
  let ans = "";
  const [n, m] = line.split(' ').map(x => parseInt(x));
  permutator(Array.from(new Array(n), (_, i) => i + 1), m)
    .forEach(arr => ans += (arr.join(' ') + '\n'));
  console.log(ans);
});

function permutator(target, m) {
  const result = [];

  const permute = (arr, acc = []) => {
    if (acc.length === m) {
      result.push(acc);
    } else {
      for (let i = 0; i < arr.length; i++) {
        const _arr = Array.from(arr);
        const next = _arr.splice(i, 1);
        if (acc.length === 0 || next > acc[acc.length - 1]) {
          permute(_arr, acc.concat(next));
        }
      }
    }
  }

  permute(target);
  return result;
}
